script/codesign/gen-trust.js

```diff
@@ -0,0 +1,38 @@
+const cp = require('child_process')
+const fs = require('fs')
+const path = require('path')
+
+const certificatePath = process.argv[2]
+const outPath = process.argv[3]
+const templatePath = path.resolve(__dirname, 'trust.xml')
+
+const template = fs.readFileSync(templatePath, 'utf8')
+
+const fingerprintResult = cp.spawnSync('openssl', ['x509', '-noout', '-fingerprint', '-sha1', '-inform', 'der', '-in', certificatePath])
+if (fingerprintResult.status !== 0) {
+  console.error(fingerprintResult.stderr.toString())
+  process.exit(1)
+}
+
+const fingerprint = fingerprintResult.stdout.toString().replace(/^SHA1 Fingerprint=/, '').replace(/:/g, '').trim()
+
+const serialResult = cp.spawnSync('openssl', ['x509', '-serial', '-noout', '-inform', 'der', '-in', certificatePath])
+if (serialResult.status !== 0) {
+  console.error(serialResult.stderr.toString())
+  process.exit(1)
+}
+
+let serialHex = serialResult.stdout.toString().replace(/^serial=/, '').trim()
+// Pad the serial number out to 18 hex chars
```

Comment:
my friend, there's an npm module for this! https://www.npmjs.com/package/left-pad


...which actually has a very helpful suggestion: [String.prototype.padStart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)